### PR TITLE
treewide: remove unzip where not needed

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Integrate changes recommended by @veprbl  in #112302

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
and you didn't remove unzip from all inputs.

File: pkgs/applications/office/ib/controller/default.nix [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
   pname = "ib-controller";
 
-  src = fetchurl {
-    url = "https://github.com/ib-controller/ib-controller/archive/${version}.tar.gz";
-    sha256 = "17a8bcgg9z3b4y38k035hm2lgvhmf8srlz59c7n2q3fdw2i95i68";
+  src = fetchFromGitHub {
+    owner = "ib-controller";
+    repo = "ib-controller";
+    rev = version;
+    sha256 = "17a8bcgg9z3b4y38k035hm2lgvhmf8srlz59c7n2q3f0w2i95i68";
```
[the remark this answers is not in the document]
Comment (siraben) [reply pkgs/applications/office/ib/controller/default.nix:11]: What's unrelated?

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/applications/office/ib/controller/default.nix]: I think he means changing the fetcher.

However fetchFromGitHub automatically unpacks the source, so it is related

**Review by veprbl — APPROVED** [review]
LGTM

**Review by jonringer — APPROVED** [review]
LGTM

failures are broken on target branch
```
https://github.com/NixOS/nixpkgs/pull/122580

9 packages marked as broken and skipped:
googleearth-pro jboss libsForQt512.soundkonverter monodevelop python38Packages.geopandas python38Packages.osmnx python39Packages.geopandas python39Packages.osmnx torchat

3 packages failed to build:
graalvm8 jvmci8 openmw-tes3mp

60 packages built:
atari800 blender btcpayserver crip debian-devscripts dotnetPackages.Nuget eventstore flacon foo2zjs fsharp41 gdal_2 git-fast-export github-runner herqq irrlicht jscoverage soundkonverter libsForQt512.herqq libsForQt514.herqq libsForQt514.soundkonverter mercurial_4 minetest minetestclient_4 minetestserver_4 minetestserver_5 msbuild mx mygui nbxplorer omnisharp-roslyn openexrid-unstable openfx openmw opentabletdriver openvdb openvino osu-lazer patool portmod prusa-slicer python-language-server python38Packages.cartopy python38Packages.fiona python38Packages.openvino python38Packages.python-simple-hipchat python38Packages.rasterio python39Packages.cartopy python39Packages.fiona python39Packages.openvino python39Packages.patool python39Packages.python-simple-hipchat python39Packages.rasterio ryujinx split2flac stuntrally super-slicer vorbisgain voxelands wasabibackend zpaq
```

**Pull request merged**